app/controllers/events_controller.rb

```diff
@@ -75,6 +85,10 @@ def generate
     render text: calendar.to_ical
   end
 
+  def show_event_highlight
+    @highlight = @event.highlights.find_by(id: params[:highlight_id])
```

Comment:
`@highlight = @event.highlights.find(params[:highlight_id])` should be more optimal